master/internal/api_user.go

```diff
@@ -325,11 +324,10 @@ func (a *apiServer) PatchUser(
 			re := regexp.MustCompile("[^\\p{Latin}\\p{N}\\s]")
 			displayName := re.ReplaceAllLiteralString(req.User.DisplayName.Value, "")
 			// Restrict 'admin' and 'determined' in display names.
-			if !(curUser.Admin && curUser.ID == uid) && strings.Contains(strings.ToLower(displayName),
-				"admin") {
+			if !targetUser.Admin && (strings.TrimSpace(strings.ToLower(displayName)) == "admin") {
 				return nil, status.Error(codes.InvalidArgument, "Non-admin user cannot be renamed 'admin'")
 			}
-			if curUser.Username != "determined" && strings.Contains(strings.ToLower(displayName),
+			if targetUser.Username != "determined" && strings.Contains(strings.ToLower(displayName),
```

Comment:
maybe have the same check as `admin`? I feel like the difference here is so subtle it'd be annoying to explain and defend it if a user asked why can't they do this.

```suggestion
			if targetUser.Username != "determined" && strings.Trimspace(strings.ToLower(displayName)) == "determined"
```

Reply:
✅  made change